test/unittests/iotools/converters/genfi_to_bids/test_genfi_to_bids_utils.py

```diff
@@ -11,8 +20,135 @@
         ("T2", "T2w"),
         ("fieldmap", "fieldmap"),
         ("fmri", "rsfmri"),
-        ("blzflbzv", None),
     ],
 )
 def test_identify_modality(input, expected):
     assert identify_modality(input) == expected
+
+
+def test_identify_modality_is_nan():
+    import math
+
+    assert math.isnan(identify_modality("blzflbzv"))
+
+
+@pytest.mark.parametrize(
+    "input, expected",
+    [
+        (
+            pd.DataFrame(
+                {
+                    "source_id": ["GRN001", "GRN001", "GRN001", "C9ORF001", "C9ORF001"],
+                    "source_ses_id": [1, 1, 1, 2, 2],
+                    "suffix": ["a", "a", "c", "a", "b"],
+                    "number_of_parts": [2, 2, 1, 1, 1],
+                    "dir_num": [10, 20, 30, 10, 20],
+                }
+            ),
+            pd.DataFrame(
+                {
+                    "source_id": ["C9ORF001", "C9ORF001", "GRN001", "GRN001", "GRN001"],
+                    "source_ses_id": [2, 2, 1, 1, 1],
+                    "suffix": ["a", "b", "a", "a", "c"],
+                    "number_of_parts": [1, 1, 2, 2, 1],
+                    "dir_num": [10, 20, 10, 20, 30],
+                    "run_01_dir_num": [10, 20, 10, 10, 30],
+                    "run": [False, False, False, True, False],
+                    "run_number": [1, 1, 1, 2, 1],
+                    "run_num": ["run-01", "run-01", "run-01", "run-02", "run-01"],
+                }
+            ),
+        )
+    ],
+)
+def test_compute_runs(input, expected):
```

Comment:
The idea of pytest's parametrize is to provide multiple inputs for which the test function will be executed. If you only have a single input and single expected value, this is overkill.
In this situation, you have basically two main options:

- If you don't need the input data for other tests, just define it within the test function and use it there (I think this is your situation here)
- If you need the data for multiple tests, you can define a fixture:

```python
@pytest.fixture
def input_df():
    return pd.DataFrame(....)

def test_compute_runs(input_df):
    # you can use input_df here

def test_compute_runs_2(input_df):
    # you can also use input_df here
```